sdks/java/extensions/sql/zetasql/src/test/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSqlDialectSpecTest.java

```diff
@@ -4067,4 +4067,46 @@ public void testSimpleTableName() {
             Row.withSchema(singleField).addValues(15L).build());
     pipeline.run().waitUntilFinish(Duration.standardMinutes(PIPELINE_EXECUTION_WAITTIME_MINUTES));
   }
+
+  @Test
+  public void testLogicalAndZetaSQL() {
+    String sql = "SELECT LOGICAL_AND(x) AS logical_and  FROM UNNEST([true, false, true]) AS x";
```

Comment:
You should also test [true, true] and [false, false]. (I'd also prefer if you cut down on code duplication by using a single [parameterized test](https://junit.org/junit4/javadoc/4.12/org/junit/runners/Parameterized.html) for all these cases.)